eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceCollectionClientProvider.cs

```diff
@@ -90,28 +142,47 @@ _getAll is null
 
         protected override PropertyProvider[] BuildProperties() => [];
 
-        private protected override IReadOnlyList<ResourceClientProvider> BuildChildResources()
-            => []; // collections should not have child resources.
-
-        protected override FieldProvider[] BuildFields() => [_clientDiagnosticsField, _clientField];
+        protected override FieldProvider[] BuildFields() => [_clientDiagnosticsField, _restClientField];
 
         protected override ConstructorProvider[] BuildConstructors()
             => [ConstructorProviderHelpers.BuildMockingConstructor(this), BuildResourceIdentifierConstructor()];
 
-        // TODO -- we need to change this type to its parent resource type.
-        private ScopedApi<ResourceType>? _resourceTypeExpression;
-        protected override ScopedApi<ResourceType> ResourceTypeExpression => _resourceTypeExpression ??= BuildCollectionResourceTypeExpression();
-
-        private ScopedApi<ResourceType> BuildCollectionResourceTypeExpression()
+        private ConstructorProvider BuildResourceIdentifierConstructor()
         {
-            // we need to know the parent of the current resource, and use the `ResourceType` property of the parent resource.
-            return Static(GetParentResourceType()).Property("ResourceType").As<ResourceType>();
+            var idParameter = new ParameterProvider("id", $"The identifier of the resource that is the target of operations.", typeof(ResourceIdentifier));
+            var parameters = new List<ParameterProvider>
+            {
+                new("client", $"The client parameters to use in these operations.", typeof(ArmClient)),
+                idParameter
```

Comment:
this is the only place we use _resourceMetadata, can we pre-calculate this in the constructor and cache the result to a field instead? the calculation is minor here, so that we can shorten the lifetime of _resourceMetadata since it could be large.